tests/mmm/test_delayed_saturated_mmm.py

```diff
@@ -580,6 +609,122 @@ def test_model_config(
             )  # beta_channel
 
 
+def new_date_ranges_to_test():
+    yield from [
+        # 2021-12-31 is the last date in the toy data
+        # Old and New dates
+        pd.date_range("2021-11-01", "2022-03-01", freq="W-MON"),
+        # Only old dates
+        pd.date_range("2019-06-01", "2021-12-31", freq="W-MON"),
+        # Only new dates
+        pd.date_range("2022-01-01", "2022-03-01", freq="W-MON"),
+        # Less than the adstock_max_lag (4) of the model
+        pd.date_range("2022-01-01", freq="W-MON", periods=1),
+    ]
+
+
+@pytest.mark.parametrize(
+    "model_name", ["mmm_fitted", "mmm_fitted_with_fourier_features"]
+)
+@pytest.mark.parametrize(
+    "new_dates",
+    new_date_ranges_to_test(),
+)
+@pytest.mark.parametrize("combined", [True, False])
+@pytest.mark.parametrize("original_scale", [True, False])
+def test_new_data_sample_posterior_predictive_method(
+    generate_data,
+    toy_X,
+    model_name: str,
+    new_dates: pd.DatetimeIndex,
+    combined: bool,
+    original_scale: bool,
+    request,
+) -> None:
+    """This is the method that is used in all the other methods that generate predictions."""
+    mmm = request.getfixturevalue(model_name)
+    X_pred = generate_data(new_dates)
+
+    posterior_predictive = mmm.sample_posterior_predictive(
+        X_pred=X_pred,
+        extend_idata=False,
+        combined=combined,
+        original_scale=original_scale,
+    )
+    pd.testing.assert_index_equal(
+        pd.DatetimeIndex(posterior_predictive.coords["date"]),
+        new_dates,
+    )
+
+    # Data has not been set back
```

Comment:
This check should now also be removed. Having a test suggests it's a feature when it's just what it is. This could be confusing for someone refactoring the code in the future because it suggests it's something that needs to be satisfied.